### ISSUE TITLE
LinkedLruHashMap: Bad state when remove() removes the last entry

### ISSUE THREAD

**Issue by hugovangalen** [google/quiver-dart#385]
My own code crashed in the method `_removeLru` with a null pointer access when calling `_tail.next`.

After debugging I discovered that the cause seems to be in the `remove()` method. 

There exists a check whether the removed `entry` is equal to `_head`, or it is equal to `_tail`. 

If it happens to be the last item that gets removed, that is not handled correctly. Right now it branches into the `entry == _head`, resets _head, but leaves _tail in its old value, which is wrong. 

If my understanding is correct, in that case both `_tail` and `_head` need to be reset to null. Adding a test for `entry == _head && entry == _tail` to reset the `_tail` and `_head` to `null` appears to fix the problem:

```
  @override
  V remove(Object key) {
    final entry = _entries.remove(key);
    if (entry != null) {
      if (entry == _head && entry == _tail) {
        _head = null;
        _tail = null;
      } else if (entry == _head) {
        _head = _head.next;
      } else if (entry == _tail) {
        _tail.previous.next = null;
        _tail = _tail.previous;
      } else {
        entry.previous.next = entry.next;
      }

      return entry.value;
    }
    return null;
  }
```

**Comment by cbracken** [comment]
Fixed by #386 and released in v0.26.1. Thanks for diagnosing and reporting this!

**Comment by hugovangalen** [comment]
@cbracken No problem. I'd like to stress that there are still issues with dangling references. See ticket #412.
